src/wopiutils.py

```diff
@@ -95,19 +106,19 @@ def generateAccessToken(userid, fileid, canedit, username, folderurl, endpoint):
       try:
         # same for MS Office, but don't try to go beyond stat
         lockInfo = _ctx['st'].stat(endpoint, getMicrosoftOfficeLockName(filename), userid)
-        canedit = False
+        viewmode = ViewMode.READ_ONLY
         locked = True
         _ctx['log'].warning('msg="Access downgraded to read-only because of an existing Microsoft Office lock" ' \
         	                  'filename="%s" mtime="%ld"' % (filename, lockInfo['mtime']))
       except IOError:
         pass
   exptime = int(time.time()) + _ctx['wopi'].tokenvalidity
-  acctok = jwt.encode({'userid': userid, 'filename': filename, 'username': username, 'canedit': canedit,
+  acctok = jwt.encode({'userid': userid, 'filename': filename, 'username': username, 'viewmode': viewmode,
                        'extlock': locked, 'folderurl': folderurl, 'exp': exptime, 'endpoint': endpoint}, \
                       _ctx['wopi'].wopisecret, algorithm='HS256').decode('UTF-8')
-  _ctx['log'].info('msg="Access token generated" userid="%s" canedit="%r" filename="%s" inode="%s" ' \
+  _ctx['log'].info('msg="Access token generated" userid="%s" mode="%s" filename="%s" inode="%s" ' \
```

Comment:
Maybe have `viewmode` in the log as well?

Reply:
I've shortened it after a first go as `viewmode` because this is how it is logged, and it's already pretty verbose:
```
2020-06-16T09:48:13 flask.app[26886] INFO     msg="Access token generated" userid="...:..." mode="ViewMode.READ_WRITE" filename="/eos/user/.../Example.odt" inode="1003128628997504" mtime="1589374408" folderurl="foo" expiration="1592380093" token="Z65sIVe-1DAZf9W2KIEc"
```

Reply:
Yes, it's verbose. But if I were to debug anything with `viewmode` I'd expect to be able to ctrl+f "viewmode" to find everything needed instead of having to figure out that I actually need to ctrl+f "mode". Ofc I can look in the code, but that takes much more time.

Reply:
...which would work as the value is `ViewMode.xxx`, thanks to the nice python representation of enums...

Reply:
But what happens if the problem would be that the value is nil?

Reply:
After the last commit, this option cannot happen - any value different from the 3 valid ones is discarded with a HTTP BAD_REQUEST.